solace-spring-cloud-stream-binder/solace-spring-cloud-stream-binder-core/src/test/java/com/solace/spring/cloud/stream/binder/util/XMLMessageMapperTest.java

```diff
@@ -1342,6 +1343,19 @@ void testMapXMLMessageToSpringMessage_WithNullPayload(boolean batchMode) {
 		}
 	}
 
+	@ParameterizedTest(name = "[{index}] batchMode={0}")
+	@MethodSource("nullPayloadXMLMessageProvider")
+	void testMapXMLMessageToSpringMessage_WhenPayloadSetAsXMLAttachment(BytesXMLMessage xmlMessage) {
+		xmlMessage.writeBytes("Hello World".getBytes()); // Write Payload as XML Attachment
+		AcknowledgmentCallback acknowledgmentCallback = Mockito.mock(AcknowledgmentCallback.class);
+		SolaceConsumerProperties consumerProperties = new SolaceConsumerProperties();
+
+		Message<?> springMessage = xmlMessageMapper.map(xmlMessage, acknowledgmentCallback, consumerProperties);
+		MessageHeaders springMessageHeaders = springMessage.getHeaders();
+
+		assertNull(springMessageHeaders.get(SolaceBinderHeaders.NULL_PAYLOAD, Boolean.class));
```

Comment:
I think you also need an assert to check that the spring message's payload is equal to `"Hello World".getBytes()`